Composer/packages/client/src/recoilModel/dispatchers/skill.ts

```diff
@@ -3,34 +3,98 @@
 /* eslint-disable react-hooks/rules-of-hooks */
 
 import { CallbackInterface, useRecoilCallback } from 'recoil';
-import { SkillManifest, SkillSetting } from '@bfc/shared';
+import { SkillManifestFile } from '@bfc/shared';
 import produce from 'immer';
 
+import { dispatcherState } from '../DispatcherWrapper';
+import { rootBotProjectIdSelector, skillsStateSelector } from '../selectors';
 import {
   skillManifestsState,
-  onAddSkillDialogCompleteState,
-  showAddSkillDialogModalState,
   displaySkillManifestState,
+  botProjectFileState,
   settingsState,
-} from './../atoms/botState';
+  botEndpointsState,
+} from '../atoms';
+
 import { setSettingState } from './setting';
 
 export const skillDispatcher = () => {
-  const createSkillManifest = ({ set }, { id, content, projectId }) => {
-    set(skillManifestsState(projectId), (skillManifests) => [...skillManifests, { content, id }]);
+  // For endpoints in manifests the settings are updated immediately in SelectSkill. If "Composer Local" is chosen needs updating when rootbot is started.
+  const updateSettingsForSkillsWithoutManifest = useRecoilCallback((callbackHelpers: CallbackInterface) => async () => {
+    const { snapshot } = callbackHelpers;
+    const botEndpoints = await snapshot.getPromise(botEndpointsState);
+    const skills = await snapshot.getPromise(skillsStateSelector);
+    const rootBotId = await snapshot.getPromise(rootBotProjectIdSelector);
+    if (!rootBotId) {
+      return;
+    }
+    const settings = await snapshot.getPromise(settingsState(rootBotId));
+    let updatedSettings = { ...settings };
+    const botProjectFile = await snapshot.getPromise(botProjectFileState(rootBotId));
+    if (!botProjectFile) {
+      return;
+    }
+
+    for (const skillNameIdentifier in botProjectFile.content.skills) {
+      const botProjectSkill = botProjectFile.content.skills[skillNameIdentifier];
+      const projectId = skills[skillNameIdentifier]?.id;
+      const currentSetting = await snapshot.getPromise(settingsState(projectId));
+
+      // Update settings only for skills that have chosen the "Composer local" endpoint and not manifest endpoints
+      if (projectId && botEndpoints[projectId] && !botProjectSkill.endpointName) {
+        updatedSettings = produce(updatedSettings, (draftState) => {
+          if (!draftState.skill) {
+            draftState.skill = {};
+          }
+          draftState.skill[skillNameIdentifier] = {
+            endpointUrl: botEndpoints[projectId],
+            msAppId: currentSetting.MicrosoftAppId ?? '',
+          };
+        });
+      }
+    }
+    setSettingState(callbackHelpers, rootBotId, updatedSettings);
+  });
+
+  const createSkillManifest = async (callbackHelpers: CallbackInterface, { id, content, projectId }) => {
+    const { set, snapshot } = callbackHelpers;
+    let manifestForBotProjectFile = undefined;
```

Comment:
Variable is undefined by default
```suggestion
    let manifestForBotProjectFile;
```